### PR TITLE
Migrate the rest of the ssl package to junit5

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We should use junit5 everywhere.

Modifications:

- Refactor rest of tests to use junit5

Result:

Part of https://github.com/netty/netty/issues/10757

### CODE REVIEW

**Comment by normanmaurer** [comment]
I think that were the last tests that use junit4! :)

File: handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java [hunk 1]
```diff
@@ -141,75 +138,88 @@ final void activate(JdkSslEngineTest instance) {
...
-                    params.add(new Object[] { providerType, bufferType, ProtocolCipherCombo.tlsv13(), true });
-                    params.add(new Object[] { providerType, bufferType, ProtocolCipherCombo.tlsv13(), false });
-                }
+    private Provider provider;
+
+    public JdkSslEngineTest() {
+        super(SslProvider.isTlsv13Supported(SslProvider.JDK));
+    }
+
+    List<JdkSSLEngineTestParam> newJdkParams() {
```
Comment (chrisvest) [inline handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java:147]: It doesn't work if this overrides `newTestParams`?

File: handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java [hunk 2]
```diff
@@ -251,65 +262,71 @@ public String select(List<String> protocols) {
...
             // java version incompatibility don't fail the test, but instead just skip the test
-            assumeNoException(e);
+            throw new AssumptionViolatedException("Not expected", e);
         }
     }
 
-    @Test
-    public void testEnablingAnAlreadyDisabledSslProtocol() throws Exception {
-        testEnablingAnAlreadyDisabledSslProtocol(new String[]{}, new String[]{ SslProtocols.TLS_v1_2 });
+    @MethodSource("newTestParams")
```
Comment (chrisvest) [inline handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java:314]: How come this one is `newTestParams` and not `newJdkParams`?

File: handler/src/test/java/io/netty/handler/ssl/SSLEngineTest.java [hunk 3]
```diff
@@ -652,19 +715,20 @@ private void testMutualAuthInvalidClientCertSucceed(ClientAuth auth) throws Exce
...
 
-        mySetupMutualAuth(serverKeyManagerFactory, commonCertChain, clientKeyManagerFactory, commonCertChain,
+        mySetupMutualAuth(param, serverKeyManagerFactory, commonCertChain, clientKeyManagerFactory, commonCertChain,
                 auth, false, false);
         assertTrue(clientLatch.await(10, TimeUnit.SECONDS));
-        assertNull(clientException);
+        rethrowIfNotNull(clientException);
         assertTrue(serverLatch.await(5, TimeUnit.SECONDS));
-        assertNull(serverException);
+        rethrowIfNotNull(serverException);
```
Comment (chrisvest) [inline handler/src/test/java/io/netty/handler/ssl/SSLEngineTest.java:723]: 👍

**Review by chrisvest — COMMENTED** [review]
Some questions

File: handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java (hunk 1 above)
Comment (normanmaurer) [reply handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java:147]: Well `newJdkParams` are only used by the tests that take `ProviderType` into account, which are only a subset of the tests.

File: handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java (hunk 2 above)
Comment (normanmaurer) [reply handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java:314]: See https://github.com/netty/netty/pull/11483#discussion_r668989254

**Comment by chrisvest** [comment]
The conscrypt test failures sound like they're related. Don't know about the interop tests with boringssl.

**Comment by normanmaurer** [comment]
@chrisvest will check

**Comment by normanmaurer** [comment]
@chrisvest ok I think I fixed it.

**Comment by normanmaurer** [comment]
This one failed again due https://github.com/netty/netty/pull/11488

**Review by chrisvest — APPROVED** [review]

**Pull request merged**